core/node/da_dispatcher/src/da_dispatcher.rs

```diff
@@ -0,0 +1,199 @@
+use std::{future::Future, time::Duration};
+
+use anyhow::Context;
+use chrono::{NaiveDateTime, Utc};
+use rand::Rng;
+use tokio::sync::watch;
+use zksync_config::DADispatcherConfig;
+use zksync_da_client::{types::DAError, DataAvailabilityClient};
+use zksync_dal::{ConnectionPool, Core, CoreDal};
+use zksync_types::L1BatchNumber;
+
+use crate::metrics::METRICS;
+
+#[derive(Debug)]
+pub struct DataAvailabilityDispatcher {
+    client: Box<dyn DataAvailabilityClient>,
+    pool: ConnectionPool<Core>,
+    config: DADispatcherConfig,
+}
+
+impl DataAvailabilityDispatcher {
+    pub fn new(
+        pool: ConnectionPool<Core>,
+        config: DADispatcherConfig,
+        client: Box<dyn DataAvailabilityClient>,
+    ) -> Self {
+        Self {
+            pool,
+            config,
+            client,
+        }
+    }
+
+    pub async fn run(self, mut stop_receiver: watch::Receiver<bool>) -> anyhow::Result<()> {
+        let pool = self.pool.clone();
+        loop {
+            if *stop_receiver.borrow() {
+                break;
+            }
+
+            tokio::join!(
+                async {
+                    if let Err(err) = self.dispatch().await {
+                        tracing::error!("dispatch error {err:?}");
+                    }
+                },
+                async {
+                    if let Err(err) = self.poll_for_inclusion(&pool).await {
```

Comment:
`pool` argument can be removed from this method as well.

Reply:
removed